src/simulation/m_rhs.fpp

```diff
@@ -1850,9 +1852,24 @@ contains
             end if
             call nvtxEndRange
         end do
+
+        if (ib) then
+            do l = 0, p
+                do k = 0, n
+                    do j = 0, m
+                        if (ib_markers%sf(j, k, l) /= 0) then
+                            do i = 1, sys_size
+                                rhs_vf(i)%sf(j, k, l) = 0d0
```

Comment:
Is it OK if this is happening on the host? I don't see any `copyin` or `copyout` near here.

Reply:
I agree. the `ib` variables may all be on the device, but without a `!$acc parallel loop` this is not happening on the device.